### PR TITLE
Fix batch operations hanging on commit and only putting last requested item

### PR DESCRIPTION
**Pull request by hinshun**
Fixes two problems:
- Currently there is a hang on batch commit because workers don't send `nil` back on `results` channel
- Batch puts and deletes only act on the last item due to variable under function closure changing as we loop

### CODE REVIEW

File: .gx/lastpubver [hunk 1]
```diff
@@ -1 +1 @@
-0.0.0: QmQTTqQnyvWoxM7gXWL1ptQ89nttDb6WsmTttnqz7G3tNf
+0.0.0: QmVF5rqMLt7sU1gfgCvwpNR4naDJ9TuvKN39WGxd9GYtx5
```
Comment (Stebalien) [inline .gx/lastpubver]: We need to bump the version before releasing. You can use `gx release patch` to do that.

File: s3.go [hunk 2]
```diff
@@ -293,9 +293,7 @@ func (b *s3Batch) Commit() error {
 	}
 
 	for _, k := range putKeys {
-		jobs <- func() error {
-			return b.s.Put(k, b.ops[k.String()].val)
-		}
+		jobs <- b.newPutJob(k, b.ops[k.String()].val)
```
Comment (Stebalien) [inline s3.go]: Ok, we _need_ tests for this package.

Comment (hinshun) [reply s3.go]: We're experimenting with a private cluster, so I'll find time to do that!

Comment (Stebalien) [reply s3.go]: That would be *awesome*. Is there any way to test s3 with mocks? My blocker here has always been a lack of access to some S3 bucket.

Comment (hinshun) [reply s3.go]: At Docker we used to test against a dockerized minio server, so we could try that. An alternative would be write a in-memory version for the AWS SDK s3 interface: https://github.com/aws/aws-sdk-go/blob/master/service/s3/s3iface/interface.go#L62

**Pull request merged**